pygit2/settings.py

```diff
@@ -59,11 +61,23 @@ class Settings:
     _search_path = SearchPathList()
 
     def __init__(self):
+        """Initialize global pygit2 and libgit2 settings."""
+        self._initialize_tls_certificate_locations()
+
+    def _initialize_tls_certificate_locations(self):
+        """Set up initial TLS file and directory lookup locations."""
         self._default_tls_verify_paths = get_default_verify_paths()
-        self.set_ssl_cert_locations(
-            self._default_tls_verify_paths.cafile,
-            self._default_tls_verify_paths.capath,
-        )
+        try:
+            self.set_ssl_cert_locations(
+                self._default_tls_verify_paths.cafile,
+                self._default_tls_verify_paths.capath,
+            )
+        except GitError as git_err:
+            valid_msg = "TLS backend doesn't support certificate locations"
+            if str(git_err) != valid_msg:
+                raise
+            self._ssl_cert_file = self._default_tls_verify_paths.cafile
```

Comment:
@jdavid is it okay like this or should we reset everything to `None`?

Reply:
I think it makes more sense to reset everything to `None`